Utilities/build-script-helper.py

```diff
@@ -8,230 +8,306 @@
+    if args.cross_compile_host:
+        if build_target == 'x86_64-apple-macosx' and args.cross_compile_host == "macosx-arm64":
+            swiftpm_args += ["--arch", "x86_64", "--arch", "arm64"]
+        elif re.match('android-', args.cross_compile_host):
+            print('Cross-compiling for %s' % args.cross_compile_host)
+            swiftpm_args += ['--destination', args.cross_compile_config]
+        else:
+            fatal_error("cannot cross-compile for %s" % args.cross_compile_host)
+
+    return swiftpm_args
+
+
+def get_swiftpm_environment_variables(swift_exec: str, args: argparse.Namespace) -> Dict[str, str]:
+    """
+    Return the environment variables that should be used for a 'swift build' or
+    'swift test' invocation.
+    """
+
+    env = dict(os.environ)
+    # Set the toolchain used in tests at runtime
+    env['SOURCEKIT_TOOLCHAIN_PATH'] = args.toolchain
+    env['INDEXSTOREDB_TOOLCHAIN_BIN_PATH'] = args.toolchain
+    # Use local dependencies (i.e. checked out next sourcekit-lsp).
+    if not args.no_local_deps:
+        env['SWIFTCI_USE_LOCAL_DEPS'] = "1"
+
+    if args.ninja_bin:
+        env['NINJA_BIN'] = args.ninja_bin
+
+    if args.sanitize and 'address' in args.sanitize:
+        # Workaround reports in Foundation: https://bugs.swift.org/browse/SR-12551
+        env['ASAN_OPTIONS'] = 'detect_leaks=false'
+    if args.sanitize and 'undefined' in args.sanitize:
+        supp = os.path.join(args.package_path, 'Utilities', 'ubsan_supressions.supp')
+        env['UBSAN_OPTIONS'] = 'halt_on_error=true,suppressions=%s' % supp
+    if args.sanitize and 'thread' in args.sanitize:
+        env['TSAN_OPTIONS'] = 'halt_on_error=true'
+
+    if args.action == 'test' and not args.skip_long_tests:
+        env['SOURCEKIT_LSP_ENABLE_LONG_TESTS'] = '1'
+
+    env['SWIFT_EXEC'] = '%sc' % (swift_exec)
+
+    return env
+
+
+def build(swift_exec: str, args: argparse.Namespace) -> None:
+    """
+    Build one product in the package
+    """
+    swiftpm_args = get_swiftpm_options(swift_exec, args)
+    env = get_swiftpm_environment_variables(swift_exec, args)
+    cmd = [swift_exec, 'build'] + swiftpm_args
+    check_call(cmd, env=env, verbose=args.verbose)
+
+
+def run_tests(swift_exec: str, args: argparse.Namespace) -> None:
+    """
+    Run all tests in the package
+    """
+    swiftpm_args = get_swiftpm_options(swift_exec, args)
+    env = get_swiftpm_environment_variables(swift_exec, args)
+
     bin_path = swiftpm_bin_path(swift_exec, swiftpm_args, env)
     tests = os.path.join(bin_path, 'sk-tests')
     print('Cleaning ' + tests)
     shutil.rmtree(tests, ignore_errors=True)
-    test_args = swiftpm_args
-    test_args += ['--parallel', '--disable-testable-imports']
-    swiftpm('test', swift_exec, test_args, env)
-  elif args.action == 'install':
+    
+    cmd = [
+        swift_exec, 'test', 
+        '--parallel', 
+        '--disable-testable-imports',
+        '--test-product', 'SourceKitLSPTests'
```

Comment:
Does this test only the `SourceKitLSPTests` target?

Reply:
Thanks for catching this, this should have been `SourceKitLSPPackageTests`, which includes all tests of the package.

Reply:
I should have probably also put this change in the PR that actually starts doing a unified build because it’s only necessary there to make sure we’re not testing the other projects as well.

Reply:
Doesn't matter to me if you want to keep it in this PR or not.

Reply:
Let’s keep it in this PR.